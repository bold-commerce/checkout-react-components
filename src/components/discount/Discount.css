.DiscountForm {
  display: flex;
}

.DiscountForm .stx-input {
  flex: 2;
}

.DiscountForm .stx-input input {
  line-height: 30px;
}

.DiscountForm button.stx-button .stx-button__text {
  text-transform: capitalize;
 }


.DiscountForm button.stx-button {
  flex: 0.5
}

.Summary__DiscountForm .DiscountForm__error {
  color: #FF1A2D;
}

@media screen and (max-width: 599px) {
  .DiscountForm {
    flex-direction: column;
  }

  .DiscountForm .stx-input {
    flex: 1;
    margin: 0 0.25rem 10px 0.25rem;

  }

  .DiscountForm button.stx-button {
    flex: 1
  }
}
