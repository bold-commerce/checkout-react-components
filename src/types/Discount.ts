export interface Discount {
  code: string,
  text: string,
  value: number,
  valid: boolean
};