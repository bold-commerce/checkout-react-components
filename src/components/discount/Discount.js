import React from 'react';
import { Input, Button } from '@bold-commerce/stacks-ui';
import useDiscount from '../../hooks/useDiscount';

import './Discount.css';

const Discount = () => {
  const {
    discount,
    setDiscount,
    discountApplied,
    errors,
    handleSubmit,
  } = useDiscount();

  return (
    <div className="SummaryBlock Summary__DiscountForm">
      <div className="DiscountForm">
        <Input
          type="text"
          placeholder="Enter discount code"
          value={discount ?? ''}
          messageText={errors && errors[0].message}
          messageType={errors && 'alert'}
          onChange={
            (e) => setDiscount(e.target.value)
          }
          disabled={discountApplied}
        />
        <Button
          primary={discountApplied || discount.length > 0}
          disabled={discount.length === 0 || discountApplied}
          onClick={() => handleSubmit()}
        >
          Apply
        </Button>
      </div>
      {
        errors && errors[0].message && (
          <div div className="DiscountForm__error">
            {
              errors[0].message
            }
          </div>
        )
      }
    </div>
  );
};

export default Discount;
