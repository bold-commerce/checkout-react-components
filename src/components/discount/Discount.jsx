/* eslint-disable react/forbid-prop-types */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Input, Button } from '@boldcommerce/stacks-ui';
import { useDiscount } from '../../hooks';
import './Discount.css';

const Discount = ({
  discountApplied, discountCode, discountErrors, applyDiscount,
}) => {
  const [discount, setDiscount] = useState(discountCode);

  return (
    <div className="SummaryBlock Summary__DiscountForm">
      <div className="DiscountForm">
        <Input
          type="text"
          placeholder="Enter discount code"
          value={discount}
          messagetext={discountErrors && discountErrors.discounts}
          messageType={discountErrors && 'alert'}
          onChange={(e) => setDiscount(e.target.value)}
          disabled={discountApplied}
        />
        <Button
          primary={discountApplied || discount.length > 0}
          disabled={discount.length === 0 || discountApplied}
          onClick={() => applyDiscount(discount)}
        >
          Apply
        </Button>
      </div>
      {
        discountErrors && discountErrors.discounts && (
          <div div className="DiscountForm__error">
            {
              discountErrors.discounts
            }
          </div>
        )
      }
    </div>
  );
};

Discount.propTypes = {
  discountApplied: PropTypes.bool,
  discountCode: PropTypes.string,
  discountErrors: PropTypes.object,
  applyDiscount: PropTypes.func,
};

const MemoizedDiscount = React.memo(Discount);

const DiscountContainer = () => {
  const {
    discountApplied, discountCode, discountErrors, applyDiscount,
  } = useDiscount();

  return (
    <MemoizedDiscount
      discountApplied={discountApplied}
      discountCode={discountCode}
      discountErrors={discountErrors}
      applyDiscount={applyDiscount}
    />
  );
};

export default DiscountContainer;
