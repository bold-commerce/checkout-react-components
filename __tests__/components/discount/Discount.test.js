import React from 'react';
import { render } from '@testing-library/react';
import { Discount } from '../../../src/components/discount/Discount';

describe('Discount', () => {
  test('renders Discount component with no discount applied', () => {
    const { asFragment } = render(
      <Discount
        discountApplied={false}
        discountCode=""
        discountErrors={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Discount component with a discount applied', () => {
    const { asFragment } = render(
      <Discount
        discountApplied
        discountCode="TESTDISCOUNT"
        discountErrors={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Discount component with invalid discount code', () => {
    const { asFragment } = render(
      <Discount
        discountApplied={false}
        discountCode="TESTDISCOUNT"
        discountErrors={{
          discounts: 'Discount code is not valid',
        }}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
