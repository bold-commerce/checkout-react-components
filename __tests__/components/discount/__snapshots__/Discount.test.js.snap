// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discount renders Discount component with a discount applied 1`] = `
<DocumentFragment>
  <div
    class="SummaryBlock Summary__DiscountForm"
  >
    <div
      class="DiscountForm"
    >
      <div
        class="stx-input stx-input--is-disabled"
      >
        <input
          aria-disabled="true"
          class="stx-input__input-element"
          disabled=""
          placeholder="Enter discount code"
          type="text"
          value="TESTDISCOUNT"
        />
      </div>
      <button
        aria-busy="false"
        aria-disabled="true"
        class="stx-button stx-button--primary stx-button--is-disabled"
        disabled=""
      >
        <span
          class="stx-button__text"
        >
          Apply
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Discount renders Discount component with invalid discount code 1`] = `
<DocumentFragment>
  <div
    class="SummaryBlock Summary__DiscountForm"
  >
    <div
      class="DiscountForm"
    >
      <div
        class="stx-input stx-input--alert"
      >
        <input
          aria-disabled="false"
          class="stx-input__input-element"
          messagetext="Discount code is not valid"
          placeholder="Enter discount code"
          type="text"
          value="TESTDISCOUNT"
        />
      </div>
      <button
        aria-busy="false"
        aria-disabled="false"
        class="stx-button stx-button--primary"
      >
        <span
          class="stx-button__text"
        >
          Apply
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
    </div>
    <div
      class="DiscountForm__error"
    >
      Discount code is not valid
    </div>
  </div>
</DocumentFragment>
`;

exports[`Discount renders Discount component with no discount applied 1`] = `
<DocumentFragment>
  <div
    class="SummaryBlock Summary__DiscountForm"
  >
    <div
      class="DiscountForm"
    >
      <div
        class="stx-input"
      >
        <input
          aria-disabled="false"
          class="stx-input__input-element"
          placeholder="Enter discount code"
          type="text"
          value=""
        />
      </div>
      <button
        aria-busy="false"
        aria-disabled="true"
        class="stx-button stx-button--is-disabled"
        disabled=""
      >
        <span
          class="stx-button__text"
        >
          Apply
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
